### DISCUSSION TITLE
Feature request: Battery preservation mode

### DISCUSSION THREAD

**Discussion by kyuzumaki** [Thomas55555/husqvarna_automower#565] [Ideas]
I recently replaced my automower battery and was thinking it would be great to have a feature to prolong the lifespan of the battery. Lithium ion cells last significantly longer if they are not cycled fully and will have a larger total lifespan throughput. I was surprised at how much of a difference this makes, you can go from 500 cycles to 6000 cycles by reducing the depth of discharge. 

My idea is just to have a setting that sends the mower out to cut and back home at certian battery levels.
So for example you might set lower battery to 40% and upper to 80%

This page has some good info on battery discharge and lifespan. It seems the upper limit has a big impact so preventing a 100% charge would be a big bonus.

https://batteryuniversity.com/article/bu-808-how-to-prolong-lithium-based-batteries

**Comment by Thomas55555** [comment]
That's not possible with the API. The mower only starts if the battery is completely full.

**Reply by kyuzumaki** [reply]
Thanks for the reply. That’s disappointing. It could make the battery last years longer.

Could send it home early but that will only have a limited effect from what I’ve read the top end is where the real longevity damage is done. 

 I wonder if Husquvarna would be open to adding the feature. Their website does not seem to have any way to submit feature suggestions.
